sdk/auth/azure_cli_authorizer.go

```diff
@@ -161,38 +191,75 @@ type azureCliConfig struct {
 
 	// DefaultSubscriptionID is the optional default subscription ID
 	DefaultSubscriptionID string
+
+	// CacheEnabled specifies whether to use module-level caching to speed up idempotent az-cli operations.
+	CacheEnabled bool
 }
 
 // newAzureCliConfig validates the supplied tenant ID and returns a new azureCliConfig.
-func newAzureCliConfig(api environments.Api, tenantId string, auxiliaryTenantIds []string) (*azureCliConfig, error) {
+func newAzureCliConfig(api environments.Api, tenantId string, auxiliaryTenantIds []string, cacheEnabled bool) (*azureCliConfig, error) {
 	var err error
 
 	// check az-cli version
-	nextMajor := azurecli.NextMajorVersion
-	if err = azurecli.CheckAzVersion(azurecli.MinimumVersion, &nextMajor); err != nil {
+	var azVersion *string
+	if cacheEnabled && cliCache.AzVersion != nil {
+		azVersion = cliCache.AzVersion
+	} else {
+		azVersion, err = azurecli.GetAzVersion()
+		if err != nil {
+			return nil, fmt.Errorf("%s. Please ensure you have installed Azure CLI version %s or newer", err, AzureCliMinimumVersion)
+		}
+
+		if cacheEnabled {
+			cliCache.AzVersion = azVersion
+		}
+	}
+	nextMajor := AzureCliNextMajorVersion
+	if err = azurecli.CheckAzVersion(*azVersion, azurecli.MsalVersion, &nextMajor); err != nil {
 		return nil, err
 	}
```

Comment:
If we make `cliCache` a pointer, we should be able to populate the entire object one-time, which'd achieve the same thing albeit a little simpler?

```suggestion
	if cliCache == nil {
		var err error
		azVersion, err = azurecli.GetAzVersion()
		if err != nil {
			return nil, fmt.Errorf("%s. Please ensure you have installed Azure CLI version %s or newer", err, AzureCliMinimumVersion)
		}
		var defaultTenantId *string
		if defaultTenantId, err = azurecli.CheckTenantID(tenantId); err != nil {
			return nil, err
		}
		if defaultTenantId == nil {
			return nil, errors.New("invalid tenantId or unable to determine tenantId")
		}
		var defaultSubscriptionId *string
		if defaultSubscriptionId, err = azurecli.GetDefaultSubscriptionID(); err != nil {
			return nil, err
		}
		cliCache = &cliCache{
			AzVersion: azVersion,
			DefaultSubscriptionId: defaultSubscriptionId,
			DefaultTenantId: defaultTenantId,
		}
	}
	
	nextMajor := AzureCliNextMajorVersion
	if err = azurecli.CheckAzVersion(*azVersion, azurecli.MsalVersion, &nextMajor); err != nil {
		return nil, err
	}
```